lib/rucio/core/rse_selector.py

```diff
@@ -133,7 +134,15 @@ def __init__(self, account, rses, weight, copies, ignore_account_limit=False, *,
         # don't consider removing rses based on the total space here - because files already on the RSE are taken into account
         # it is possible to have no space but still be able to fulfil the rule
 
-    def select_rse(self, size, preferred_rse_ids, copies=0, blocklist=[], prioritize_order_over_weight=False, existing_rse_size=None):
+    def select_rse(
+        self,
+        size: int,
+        preferred_rse_ids: Iterable[str],
+        copies: int = 0,
+        blocklist: Optional[list[str]] = None,
```

Comment:
`Sequence` instead of `str`

Reply:
(Assuming you mean `Sequence` instead of `list` here as well)

Changed to:
```
blocklist: Optional[Sequence[str]] = None,
```